client/serviceregistration/checks/client_test.go

```diff
@@ -269,6 +271,25 @@ func TestChecker_Do_HTTP_extras(t *testing.T) {
 				[2]string{"Authorization", "Basic ZWxhc3RpYzpjaGFuZ2VtZQ=="},
 			),
 		},
+		{
+			name:   "host header",
+			method: "GET",
+			headers: makeHeaders(encoding, agent,
+				[2]string{"Host", "hello"},
+				[2]string{"Test-Abc", "hello"},
+			),
+		},
+		{
+			name:   "host header",
```

Comment:
nit: this is the second test case named `"host header"`; making them unique helps quickly tracking down failures